components/eam/src/physics/cam/microp_aero.F90

```diff
@@ -646,10 +658,35 @@ subroutine microp_aero_run ( &
     
    call outfld('LCLOUD', lcldn, pcols, lchnk)
 
+   ! Init qqcw (a flat 1d pointer array) to mode number and species mass mixing ratios for
+   ! cloud borne phases
+
+   allocate(qqcw(ncnst_tot)) 
```

Comment:
We can deallocate this array after the call to dropmixnuc. Below is the preferred solution but we can stick with what you have as it is similar to the original code.

A better solution would involve a few code changes:
Since `ncnst_tot` is computed at model initialization, we can allocate this array in `microp_aero_init` after the call to `call ndrop_init()`.  Before pointing derived type `qqcw` with the valid values in the `microp_aero_run` subroutine (before calling `dropmixnuc`), we can initialize it to `huge(1.0_r8)` so that it doesn't carry old values.

This way, the allocation will only happen once only during the model initialization. We may also need to declare `qqcw` thread private as it will become shared if we declare `qqcw` at the module level.

Reply:
I've kept the version more like the original code for now, but I do dellocate qqcw after the call to dropmixnuc.